src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy

```diff
@@ -28,148 +28,141 @@ import java.nio.file.attribute.PosixFilePermissions
 class RoddyIOHelperMethods {
 
 
-    public static abstract class Compressor {
-        public void compress(File from, File to, File workingDirectory) {
+    static abstract class Compressor {
+
+        void compress(File from, File to) {
             if (from.isFile())
-                compressFile(from, to, workingDirectory);
+                compressFile(from, to)
 
             else if (from.isDirectory())
-                compressDirectory(from, to, workingDirectory);
-        }
-
-        public void compress(File from, File to) {
-            compress(from, to, null);
+                compressDirectory(from, to)
         }
 
-        public abstract void decompress(File from, File to);
+        @Deprecated
+        void decompress(File from, File to, File wd) { decompress(from, to) }
 
-        public abstract void decompress(File from, File to, File workingDirectory);
+        abstract void decompress(File from, File to)
 
-        public abstract void compressFile(File from, File to);
+        @Deprecated
+        void compressFile(File from, File to, File wd) { compressFile(from, to) }
 
-        public abstract void compressFile(File from, File to, File workingDirectory);
+        abstract void compressFile(File from, File to)
 
-        public abstract void compressDirectory(File from, File to);
+        @Deprecated
+        void compressDirectory(File from, File to, File wd) { compressDirectory(from, to) }
 
-        public abstract void compressDirectory(File from, File to, File workingDirectory);
+        abstract void compressDirectory(File from, File to)
 
-        public abstract GString getCompressionString(File from, File to);
+        @Deprecated
+        GString getCompressionString(File from, File to, File wd) { getCompressionString(from, to) }
 
-        public abstract GString getDecompressionString(File from, File to);
+        abstract GString getCompressionString(File from, File to)
 
-        public abstract GString getCompressionString(File from, File to, File workingDirectory);
+        @Deprecated
+        GString getDecompressionString(File from, File to, File wd) { getDecompressionString(from, to) }
 
-        public abstract GString getDecompressionString(File from, File to, File workingDirectory);
+        abstract GString getDecompressionString(File from, File to)
     }
 
     /**
      * Produces md5 compatible zipped archives of the input file / folder
      * Also produces a file named [zip.gz].md5
      */
-    public static class NativeLinuxZipCompressor extends Compressor {
+    static class NativeLinuxZipCompressor extends Compressor {
```

Comment:
Why "Native"? Why "Linux? Why not "Bash"? Or "Posix"? What about `-d` and `-f` and the `if` syntax? Why make this a nested class? Should this not just all go into a "compressor" package? 

Sorry for all the questions.